src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -112,7 +121,7 @@ func newUserInteractiveResponse(
 	params map[string]interface{},
 ) userInteractiveResponse {
 	return userInteractiveResponse{
-		fs, sessions[sessionID], params, sessionID,
+		fs, sessions.Get(sessionID), params, sessionID,
```

Comment:
My only concern is if a developer doesn't know to use `.Get()` and instead just writes an `if session, ok := sessionsDict; ok {...}`, but this would likely be picked up in review anyway.

Reply:
Making `sessionsDict` an actual struct to stop people accessing the map directly. It looks like the only other operation that is done is to append a `authtypes.LoginType` to a session, so should be a small interface to implement.

Reply:
@erikjohnston Should I go ahead and do that ?

Reply:
Sure, if you're happy